src/d_delay.c

```diff
@@ -97,16 +103,24 @@ static t_int *sigdelwrite_perform(t_int *w)
     t_sample *in = (t_sample *)(w[1]);
     t_delwritectl *c = (t_delwritectl *)(w[2]);
     int n = (int)(w[3]);
+    t_sigdelwrite *x = (t_sigdelwrite *)(w[4]);
     int phase = c->c_phase, nsamps = c->c_n;
     t_sample *vp = c->c_vec, *bp = vp + phase, *ep = vp + (c->c_n + XTRASAMPS);
     phase += n;
 
     while (n--)
```

Comment:
would it be better to put the freeze check outside of the while loop?
like:

(outside of sigdelwrite_perform):
```
static void sigdelwrite_wrapphase(t_sample **loc, t_sample *vector, t_sample *end, int *phase, int nsamps)
{
    if(*loc == end)
    {
        vector[0] = end[-4];
        vector[1] = end[-3];
        vector[2] = end[-2];
        vector[3] = end[-1];
        *loc = vector + XTRASAMPS;
        *phase -= nsamps;
    }
}
```
then in perform:
```
if(x->x_freeze)
{
    while(n--)
    {
        bp++;
        sigdelwrite_wrapphase(&bp, vp, ep, &phase, nsamps);
    }
}
else
{
    while(n--)
    {
        t_sample f = *in++;
        if (PD_BIGORSMALL(f))
            f = 0;
        *bp++ = f;
        sigdelwrite_wrapphase(&bp, vp, ep, &phase, nsamps);
    }
}
```
just a thought, not sure if it's worth it or if the compiler would do that anyways

Reply:
maybe you can have an alternative PR and if it's considered best I'll remove this

Reply:
@porres maybe I'll just try to add commits to this.. that seems cleaner

Reply:
let's wait to hear the usual wizard like @Spacechild1 and @umlaeute

Reply:
Usually, the compiler can hoist a check out of the loop if it can prove that the value won't change. This is very likely the case here (for an explanation, you can google "strict aliasing rule").

If you want to make sure, you can cache `x->x_freeze` in a local variable and no sane compiler would do the check on every iteration.